RELEASING.md

```diff
@@ -0,0 +1,241 @@
+# Versioning and release processes
+
+## Versioning Process
+
+Ledger team has fully adopted the [PVP versioning scheme](https://pvp.haskell.org/) for
+all packages in this repo.
+
+Updating the version for every package in this repository is the responsinility of the
+developer that is introducing a change to any package that is subject to release
+process. Version must be updated in both the `CHANGELOG.md` and in the `.cabal` file for
+the package in the same PR, as well as every other package that is affected by the change
+in the repository. Every affected package that is subject to release process must receive
+a lower and, potentially, an upper bound update on the dependency that has experienced the
+change.
+
+For example if I add a function to `cardano-ledger-core` and use it in
+`cardano-ledger-conway`, then I must add a minor version bump to `cardano-ledger-core` and
+I also must add a lower version restriction in `cardano-ledger-conway` for
+`cardano-ledger-core`, as per PVP.
+
+In other words, when it comes to versioning, the fact that all of the packages in this
+repository are build together at the same time should be ignored. It should be assumed
+that every package can be released to CHaPs by itself at any point in time. This is
+necessary to guarantee that if such a package is compatible with other packages from this
+repository, which have been previously released, the version bounds should reflect that.
+
+### Upper bounds
+
+Normally, [strict upper bounds are required by
+PVP](https://pvp.haskell.org/#dependencies-in-cabal) for all dependencies of a
+package. This is the only exception we make to the PVP, and we do not require strict upper
+bounds for dependenicies that come from Hackage. For exmaple there is no need to add an
+upper bound on `containers` package, because that's just dumb. On the other hand all of
+the dependeices that come from CHaPs should follow strict bounds, because those packages
+are a lot more volatile and creating revisions for packages on CHaPs is not as smooth as
+it is on Hackage.
+
+### `CHANGELOG.md`
+
+Every package, that [is subject to the release
+process](packages-excluded-from-release-process), will always have the top most entry in
+the `CHANGELOG.md` set to an unreleased version. That top most section will have either
```

Comment:
Did we agree to putting "(unreleased)" next to the top entry?

For example:

```
1.2.0 (unreleased)
* stuff

1.1.0
* things
```

This way, when `1.2.0` is released, we do not need to do something like add a new `1.2.1` or some such as a part of release process (to maintain the invariant that we "always have the top most entry in the `CHANGELOG.md` set to an unreleased version").

Reply:
chatted in a call, I like the new plan!